lib/ecto/query/builder/from.ex

```diff
@@ -58,15 +60,13 @@ defmodule Ecto.Query.Builder.From do
   If possible, it does all calculations at compile time to avoid
   runtime work.
   """
-  @spec build(Macro.t(), Macro.Env.t(), atom, String.t | nil, nil | {:ok, String.t | nil} | [String.t]) ::
+  @spec build(Macro.t(), Macro.Env.t(), atom, {:ok, String.t | nil} | nil, hints | nil) ::
           {Macro.t(), Keyword.t(), non_neg_integer | nil}
-  def build(query, env, as, prefix, maybe_hints) do
-    hints = List.wrap(maybe_hints)
-
+  def build(query, env, as, prefix, hints) do
     unless Enum.all?(hints, &is_valid_hint/1) do
       Builder.error!(
-        "`hints` must be a compile time string, list of strings, or a tuple " <>
-          "got: `#{Macro.to_string(maybe_hints)}`"
+        "`hints` must be a list of compile time strings or tuples " <>
```

Comment:
The previous warning was correct. We changed the implementation but the user API is the same!

Reply:
🙈

I got carried away.

Reply:
Fixed.